Add advice of using `assertEquals()` in the contributing guide (#5594)
http4s/http4s: website/src/hugo/content/contributing.md

* We encourage the addition of arbitrary instances to
  `org.http4s.testing.Http4sArbitraries` to support richer property
  testing.

> * For assertions in tests using `assertEquals()` more preferable than `assert()`. 

Reviewer: ```suggestion
* For assertions in tests using `assertEquals(a, b)` is preferable to `assert(a == b)`. 
```

Author: Thanks!
